k8s/migration/internal/controller/migrationplan_controller.go

```diff
@@ -541,6 +539,12 @@ func (r *MigrationPlanReconciler) CreateMigrationConfigMap(ctx context.Context,
 				"HEALTH_CHECK_PORT":     migrationplan.Spec.MigrationStrategy.HealthCheckPort,
 			},
 		}
+		// Check if target flavor is set
+		if vmMachine.Spec.TargetFlavorId != "" {
```

Comment:
Move the logic of finding next best flavor to the controller, this way we can maintain single source of truth for flavors